authorization-server/utils/id-token-cache.js

```diff
@@ -1,9 +1,23 @@
 const cache = {};
 
-export function cache_id_token(subject, id_token) {
-  cache[subject] = id_token;
+/**
+ *
+ * Caches the ID Assertion after SSO for token exchange
+ *
+ * NOTE: In production, the cache lookup key should be the authorization_code
+ *
+ */
+
+function makeCacheKey(subject, client_id) {
+  return [subject, client_id].join('|');
+}
+
+export function cacheSubjectToken(subject, client_id, subjectToken, tokenType) {
+  cache[makeCacheKey(subject, client_id)] = { subjectToken, tokenType };
 }
 
-export function getIdToken(subject) {
-  return cache[subject];
+export function getSubjectToken(subject, client_id) {
+  return (
+    cache[makeCacheKey(subject, client_id)] || { subjectToken: undefined, tokenType: undefined }
```

Comment:
```suggestion
    cache[makeCacheKey(subject, client_id)] ?? { subjectToken: undefined, tokenType: undefined }
```

Reply:
Note: I would prefer `null` instead of `undefined` if we don't get issues with that.

Reply:
Also maybe this would work?

```suggestion
    cache[makeCacheKey(subject, client_id)] ?? {}
```